File: backend/internal/models/sharing.go
package models

import "time"

type UserPairingCode struct {
	ID        uint      `gorm:"primaryKey" json:"id"`
	UserID    uint      `gorm:"not null;uniqueIndex" json:"user_id"`
	Code      string    `gorm:"not null;uniqueIndex;size:9" json:"code"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}

func (UserPairingCode) TableName() string {
	return "user_pairing_codes"
}

type SharedAccess struct {
	ID        uint      `gorm:"primaryKey" json:"id"`
	OwnerID   uint      `gorm:"not null;index" json:"owner_id"`
	ViewerID  uint      `gorm:"not null;index" json:"viewer_id"`
	CreatedAt time.Time `json:"created_at"`

	Owner  User `gorm:"foreignKey:OwnerID" json:"owner,omitempty"`
	Viewer User `gorm:"foreignKey:ViewerID" json:"viewer,omitempty"`
}

func (SharedAccess) TableName() string {
	return "shared_access"
}

File: frontend/src/types/sharing.ts
export interface PairingCodeResponse {
  code: string
}

export interface SharedAccess {
  id: number
  owner_id: number
  viewer_id: number
  created_at: string
  owner?: { id: number; email: string; name: string }
  viewer?: { id: number; email: string; name: string }
}

export interface ConnectionsResponse {
  viewers: SharedAccess[]
  owners: SharedAccess[]
}
